lib/improver/tests/calculate_sleet_prob/test_calculate_sleet_probability.py

```diff
@@ -0,0 +1,106 @@
+# -*- coding: utf-8 -*-
+# -----------------------------------------------------------------------------
+# (C) British Crown Copyright 2017-2019 Met Office.
+# All rights reserved.
+#
+# Redistribution and use in source and binary forms, with or without
+# modification, are permitted provided that the following conditions are met:
+#
+# * Redistributions of source code must retain the above copyright notice, this
+#   list of conditions and the following disclaimer.
+#
+# * Redistributions in binary form must reproduce the above copyright notice,
+#   this list of conditions and the following disclaimer in the documentation
+#   and/or other materials provided with the distribution.
+#
+# * Neither the name of the copyright holder nor the names of its
+#   contributors may be used to endorse or promote products derived from
+#   this software without specific prior written permission.
+#
+# THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+# AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+# IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+# ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE
+# LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+# CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+# SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+# INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+# CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+# ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+# POSSIBILITY OF SUCH DAMAGE.
+"""Unit tests for the calculate_sleet_probability plugin."""
+
+import unittest
+
+import numpy as np
+from iris.tests import IrisTest
+from improver.calculate_sleet_prob import calculate_sleet_probability
+from improver.tests.set_up_test_cubes import set_up_probability_cube
+
+
+class Test_calculate_sleet_probability(IrisTest):
+    """ Tests the calculate sleet probability function."""
+
+    def setUp(self):
+        """Create cubes to input into the function."""
+
+        self.thresholds = np.array([276, 277], dtype=np.float32)
+
+        rain_prob = np.array([[[0.5, 0.1, 1.0],
+                               [0.0, 0.2, 0.5],
+                               [0.1, 0.1, 0.3]],
+                              [[0.5, 0.1, 1.0],
+                               [0.0, 0.2, 0.5],
+                               [0.1, 0.1, 0.3]]], dtype=np.float32)
+        self.rain_prob_cube = set_up_probability_cube(
+            rain_prob, self.thresholds)
```

Comment:
Before this gets merged it would be nice to give these cubes suitable names. I know it doesn't matter in the grand scheme of things, but it's always nice to test with data that looks like that data we will really be using. Again, I'm not exactly sure what the names will be, so that's one for Stephen as well, e.g.
```
self.rain_prob_cube = set_up_probability_cube(
    rain_prob, self.thresholds, variable_name='probability_of_altitude_of_rain_falling_level_at_surface')
```